credential/builder.go

```diff
@@ -33,19 +33,25 @@ type VerifiableCredentialBuilder struct {
 
 // NewVerifiableCredentialBuilder returns an initialized credential builder with some default fields populated
 // Default id is empty
-func NewVerifiableCredentialBuilder() VerifiableCredentialBuilder {
+func NewVerifiableCredentialBuilder(emptyID ...bool) VerifiableCredentialBuilder {
```

Comment:
what do you think about making this a bit more readabile?

we could change the build to accept a new type like `IDvalue` for example


```go

type IDValue string

const EmptyIDValue IDValue = ""

func NewVerifiableCredentialBuilder(id IDValue) VerifiableCredentialBuilder {
    if id == EmptyIDValue {
        // generate ID
    }
    ...
}
```

Reply:
Hi @decentralgabe, thanks for your inputs. 
Please confirm if the behaviour is as follows: 
```
NewVerifiableCredentialBuilder (EmptyIDValue)  -> creates vc with id as uuid
NewVerifiableCredentialBuilder  ("xxx") -> generates vc with id as "xxx"
vcb.SetID("") ->  sets ID as empty
```

Reply:
ah, good consideration.

Perhaps we should add....

`EmptyIDValue` -> set ID value to empty
`GenerateIDValue` -> generate a uuid
`CustomIDValue("<your-id-here>")` -> use custom ID

to be unambiguous?

Reply:
Hi @decentralgabe , I have added IDValue argument here to all three cases (empty ID/ generate ID/ custom ID)